### PR TITLE
Update PT 1.9 to remove repo.anaconda.com

### PR DESCRIPTION
**Pull request by kevinyang8**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Update PT 1.9 to remove repo.anaconda.com

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by tejaschumbalkar — COMMENTED** [review]
LGTM

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py [hunk 1]
```diff
@@ -170,6 +170,10 @@ def test_smmodelparallel_mnist_multigpu_singlenode(ecr_image, instance_type, sag
     """
     Tests pt gpt2 command via script mode
     """
+    _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+    if Version("1.9") == Version(image_framework_version):
+        pytest.skip("Test is not supported on PyTorch 1.9")
```
Comment (saimidu) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py:175]: See line 45 for a better type of version check, where you can check all patch versions of PT 1.9 between "1.9" and "1.10", rather than specifically comparing to "1.9".

With this `==` method of comparison, the condition will always be ignored for versions "1.9.1" or "1.9.2", etc.

Comment (kevinyang8) [reply test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py]: This fixed the problem, thanks

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smdataparallel.py [hunk 2]
```diff
@@ -93,6 +93,10 @@ def test_smdataparallel_mnist_script_mode_multigpu(ecr_image, sagemaker_regions,
     """
     validate_or_skip_smdataparallel(ecr_image)
 
+    _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+    if Version("1.9") <= Version(image_framework_version) < Version("1.10"):
+        pytest.skip("Test is not supported on PyTorch 1.9")
```
Comment (saimidu) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smdataparallel.py:98]: This is not an SM ModelParallel test.

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smdataparallel.py [hunk 3]
```diff
@@ -148,6 +152,10 @@ def test_smmodelparallel_smdataparallel_mnist(instance_types, ecr_image, sagemak
     This test has been added for SM DataParallelism and ModelParallelism tests for re:invent.
     TODO: Consider reworking these tests after re:Invent releases are done
     """
+    _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+    if Version("1.9") <= Version(image_framework_version) < Version("1.10"):
```
Comment (saimidu) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smdataparallel.py:156]: Please use a `SpecifierSet` instead of comparing to two versions.

```suggestion
    if Version(image_framework_version) in SpecifierSet(">=1.9,<1.10"):
```
